core/market/src/negotiation/common.rs

```diff
@@ -295,6 +379,21 @@ impl CommonBroker {
         }
         Ok(())
     }
+
+    pub async fn notify_agreement(&self, agreement: &Agreement) {
+        let session_notifier = self.session_notifier.clone();
+
+        // Notify everyone waiting on Agreement events endpoint.
+        if let Some(_) = &agreement.session_id {
```

Comment:
Yes we do. But it doesn't hurt, because this is only notification and `query_agreement_events` must get events from database whatsoever. So it won't find any events for wrong identity.
But I modified comment near: https://github.com/golemfactory/yagna/pull/793#discussion_r528020727 to make it explicit